Makefile

```diff
@@ -17,6 +17,9 @@ build: require-INSTALL_TO
 test: require-INSTALL_TO
 	PATH=${INSTALL_TO}/:${PATH} stack test --system-ghc --bench
 
+doc: require-INSTALL_TO
+	PATH=${INSTALL_TO}/:${PATH} stack build --haddock --haddock-deps --haddock-internal --haddock-hyperlink-source
```

Comment:
We probably don't need to update `PATH` here, that's only been necessary to ensure CI can find `consul`, for tests. Does the doc build depend on `consul`?